src/radix_connect/mobile/relay_service/models/response/session_handshake_request.rs

```diff
@@ -0,0 +1,61 @@
+use crate::prelude::*;
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct SessionHandshakeRequest {
+    #[serde_as(as = "DisplayFromStr")]
+    pub public_key: PublicKey,
+}
+
+impl SessionHandshakeRequest {
+    pub fn new(public_key: impl Into<PublicKey>) -> Self {
+        Self {
+            public_key: public_key.into(),
+        }
+    }
+}
+
+impl HasSampleValues for SessionHandshakeRequest {
+    fn sample() -> Self {
+        Self::new(PublicKey::sample())
+    }
+
+    fn sample_other() -> Self {
+        Self::new(PublicKey::sample_other())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = SessionHandshakeRequest;
+
+    #[test]
+    fn equality() {
+        assert_eq!(SUT::sample(), SUT::sample());
+        assert_eq!(SUT::sample_other(), SUT::sample_other());
+    }
+
+    #[test]
+    fn inequality() {
+        assert_ne!(SUT::sample(), SUT::sample_other());
+    }
+
+    #[test]
+    fn json_roundtrip() {
+        let original = SUT::sample();
+        let json = format!(
+            r#"
+            {{
+                "publicKey":"{}"
```

Comment:
extra important here to NOT use `format! because if we change our impl of `Display` of `PublicKey`, this test will just work, but we might have broken Sargon!